codegen/src/generator/json.rs

```diff
@@ -5,14 +5,23 @@ use super::GenerateProtocol;
 
 pub struct JsonGenerator;
 
+fn generate_result_type<'a>(service: &Service, operation: &Operation, output_type: &'a str) -> String {
```

Comment:
This is probably just me being **really** nitpicky here, but any reason why this generator function is above the `impl` while the rest are below?